### PR TITLE
[BCHDCC-50] Add Translation Bar with Flag Icons to Header

### PR DESCRIPTION
**Pull request by emmabjj**
## Description

Add responsive, accessible translation bar with flag icons to header and hamburger menu. English, Arabic, Chinese Simplified, Chinese Traditional, French, Korean, and Spanish. Flag icon links correctly translate the page into the language chosen.

Followed this [figma design](https://www.figma.com/design/xli1XXrftaijx1KLldloK7/BCHD---CHARMcare%3A-Tailwind-Upgrade?node-id=2389-2504&t=UbP7xBuoeVwNMWtM-0)

## Motivation and Context

[BCHDCC-50](https://app.clickup.com/t/9006094761/BCHDCC-50) - Add a "Translate to" bar under the main header - full page layout 

This translation bar replaces the google translate widget in the non-admin site. 

### Desktop (translation bar under header)
Screenshot 2024-09-11 at 5 00 02 PM

-------------------------

### iPad (in hamburger)
Screenshot 2024-09-11 at 5 02 08 PM

-------------------------

### Mobile (in hamburger)
Screenshot 2024-09-11 at 5 02 35 PM

-------------------------

#### Arabic
Screenshot 2024-09-11 at 5 03 52 PM

#### French
Screenshot 2024-09-11 at 5 04 21 PM

#### Chinese
Screenshot 2024-09-11 at 5 04 58 PM

#### Spanish
Screenshot 2024-09-11 at 5 05 47 PM

**Libraries Added**
No libraries added

## Tests to Run
All existing tests pass

## Internal Checklist

- [x] My code follows the code style of this project (https://rubystyle.guide/).
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: app/views/shared/_header.html.haml [hunk 1]
```diff
@@ -45,5 +61,81 @@
...
+        }
+        if (element.childNodes.length === 1 && element.childNodes[0].nodeType === Node.TEXT_NODE) {
+          translateElement(element, lang);
+        }
+      }
+      currentLang = lang;
+    }
+
+    function translateElement(element, lang) {
+      var text = element.textContent || element.innerText;
```
Comment (anna-dorigo) [inline app/views/shared/_header.html.haml:107]: While testing this locally I noticed that the placeholders are not translated: 
image

image

Maybe we need to add here a case to account for the placeholder before we sent the text to the API.

**Review by anna-dorigo — CHANGES_REQUESTED** [review]
This looks great on all the different browser sizes, and the menu looks clean and nice. There is just a small issue that I pointed put.

File: app/views/shared/_header.html.haml (hunk 1 above)
Comment (emmabjj) [reply app/views/shared/_header.html.haml:107]: @anna-dorigo Just added a case for the placeholder text in the js - thanks! 👏

**Review by anna-dorigo — APPROVED** [review]
👍

**Pull request merged**